### PR TITLE
[Patch] New rule to lint duplicate attributes

### PR DESCRIPTION
**Pull request by siva-sundar**
Throws warning when a duplicate attribute is detected in any of the following

- ElementNode
- MustacheStatement
- BlockStatement
- SubExpression

### CODE REVIEW

File: lib/rules/index.js [hunk 1]
```diff
@@ -18,5 +18,6 @@ module.exports = {
   'invalid-interactive': require('./lint-invalid-interactive'),
   'style-concatenation': require('./lint-style-concatenation'),
   'simple-unless': require('./lint-simple-unless'),
-  'unused-block-params': require('./lint-unused-block-params')
+  'unused-block-params': require('./lint-unused-block-params'),
+  'duplicate-attributes': require('./lint-duplicate-attributes')
```
Comment (rwjblue) [inline lib/rules/index.js]: Let's call this "no-duplicate-attributes"?

File: lib/rules/lint-duplicate-attributes.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const Rule = require('./base');
+
+const logDuplicateAttributes = function ({ attributes = [], indentifier, type }) {
```
Comment (rwjblue) [inline lib/rules/lint-duplicate-attributes.js]: Can't use destructuring (we still support node 4).

File: lib/rules/lint-duplicate-attributes.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  attributes.forEach((attribute, index) => {
+    for (currentIndex = index; currentIndex < length; currentIndex++) {
+      currentAttribute = attributes[currentIndex];
+      if (attribute[indentifier] === currentAttribute[indentifier] && index !== currentIndex) {
+        this.log({
+          message: `Duplicate attributes found in the ${type}.`,
+          line: currentAttribute.loc && currentAttribute.loc.start.line,
+          column: currentAttribute.loc && currentAttribute.loc.start.column,
+          source: this.sourceForNode(currentAttribute)
```
Comment (rwjblue) [inline lib/rules/lint-duplicate-attributes.js]: I think the source should be the node containing the attribute (not the attribute itself), what do you think?

File: lib/rules/index.js (hunk 1 above)
Comment (siva-sundar) [inline lib/rules/index.js]: Sure 👍

File: lib/rules/lint-duplicate-attributes.js (hunk 3 above)
Comment (siva-sundar) [inline lib/rules/lint-duplicate-attributes.js]: Yeah, that will enable us to locate the code much quicker. Can we log as I have mentioned below ?

`Duplicate attribute 'name' found in the SubExpression. `
`(hash name=name age=age name=name) `

**Comment by siva-sundar** [comment]
Should this be`recommended` rule ?

**Comment by rwjblue** [comment]
Yes, it is a good idea to make it recommended but doing so would be a breaking change. 

Let's land and release the new rule, then create an issue to update the recommended config for our next major version bump.

File: lib/rules/lint-duplicate-attributes.js (hunk 3 above)
Comment (siva-sundar) [inline lib/rules/lint-duplicate-attributes.js]: I have added logs as mentioned above. Is it ok?

**Comment by siva-sundar** [comment]
👍

**Comment by siva-sundar** [comment]
I have done the changes you have mentioned. Let me know if anything else needs to be done

File: lib/rules/lint-duplicate-attributes.js (hunk 3 above)
Comment (rwjblue) [inline lib/rules/lint-duplicate-attributes.js]: Yes, this looks good. 😸

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**